src/app/loans/add-loan-charge/add-loan-charge.component.ts

```diff
@@ -0,0 +1,108 @@
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { LoansService } from '../loans.service';
+
+/**
+ * Create Add Loan Charge component.
+ */
+@Component({
+  selector: 'mifosx-addloancharge',
+  templateUrl: './add-loan-charge.component.html',
+  styleUrls: ['./add-loan-charge.component.scss']
+})
+export class AddLoanChargeComponent implements OnInit {
+
+  /** Minimum Due Date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum Due Date allowed. */
+  maxDate = new Date();
+  /** Add Loan Charge form. */
+  loanChargeForm: FormGroup;
+  /** loan charge options. */
+  loanChargeOptions: {
+    id: number;
+    name: string;
+    amount: number;
+    chargeCalculationType: {
+      value: any;
+    };
+    chargeTimeType: {
+      id: number;
+      value: any;
+    };
+  }[];
+  /** loan Id of the loan account. */
+  loanId: string;
+
+  /**
+   * Retrieves the loan charge template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {AccountingService} accountingService Accounting Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe,
+              private loansService: LoansService) {
+    this.route.data.subscribe((data: { loanChargeTemplate: any }) => {
+      this.loanChargeOptions = data.loanChargeTemplate.chargeOptions;
+    });
+    this.loanId = this.route.snapshot.params['loanId'];
+  }
+
+  /**
+   * Creates the Loan Charge form.
+   */
+  ngOnInit() {
+    this.createLoanChargeForm();
+    this.loanChargeForm.controls.chargeId.valueChanges.subscribe(chargeId => {
+      const chargeDetails = this.loanChargeOptions.find(option => {
+        return option.id === chargeId;
+      });
+      if (chargeDetails.chargeTimeType.id === 2) {
+        this.loanChargeForm.addControl('dueDate', new FormControl('', Validators.required));
+      } else {
+        this.loanChargeForm.removeControl('dueDate');
+      }
+      this.loanChargeForm.patchValue({
+        'amount': chargeDetails.amount,
+        'chargeCalculation': chargeDetails.chargeCalculationType.value,
+        'chargeTime': chargeDetails.chargeTimeType.value
+      });
+    });
+  }
+
+  /**
+   * Creates the Loan Charge form.
+   */
+  createLoanChargeForm() {
+    this.loanChargeForm = this.formBuilder.group({
+      'chargeId': ['', Validators.required],
+      'amount': ['', Validators.required],
+      'chargeCalculation': [{ value: '', disabled: true }],
+      'chargeTime': [{ value: '', disabled: true }]
+    });
+  }
+
+  submit() {
+    const prevDueDate: Date = this.loanChargeForm.value.dueDate;
+    // TODO: Update once language and date settings are setup
+    const dateFormat = 'yyyy-MM-dd';
+    this.loanChargeForm.patchValue({
```

Comment:
@Jov03 This seems a nicer way to implement what we have been doing. Could you check and let me know if you are able to update the wiki to reflect the same?

Reply:
@abhaychawla i am unable to understand which wiki you are referring to??

Reply:
Here: https://github.com/openMF/web-app/wiki/Date-and-Language

Reply:
@abhaychawla contributors don't have rights to edit wiki pages  so i cannot edit the same.

Reply:
Alright, let me see what can be done.

Reply:
@Jov03 Please accept the collaborator invite and see if its accessible now.

Reply:
@abhaychawla i have updated the wiki to reflect the same please have a look.

Reply:
👍